repository/users/user.go

```diff
@@ -0,0 +1,17 @@
+package users
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/ivanterekh/go-skeleton/model"
+)
+
+// ErrNoSuchUser is the error returned by
+// functions that search for users.
+var ErrNoSuchUser = errors.Errorf("no such user")
+
+// Repository is an interface for user searching.
+type Repository interface {
```

Comment:
I'm not sure that we need this interface?

Reply:
Now it is used to easily switch between mock repository and sql db. And it probably could be useful in future in case of switching to the other db.

Reply:
in this case, it's `AuthRepository` or `UserRepository`